docs/developer-guide/creating-protobuf-schemas.md

```diff
@@ -87,23 +89,43 @@ The Substreams CLI is used to generate the associated Rust code for the protobuf
 {% endhint %}
 
 {% code overflow="wrap" %}
+
 ```bash
 substreams protogen ./substreams.yaml --exclude-paths="sf/ethereum,sf/substreams,google"
 ```
+
 {% endcode %}
 
 The Rust code is generated and saved into [`src/pb/eth.erc721.v1.rs`](https://github.com/streamingfast/substreams-template/blob/develop/src/pb/eth.erc721.v1.rs)``
 
 The [`mod.rs`](https://github.com/streamingfast/substreams-template/blob/develop/src/pb/mod.rs) file located in the `src/pb` directory of the Substreams Template example is responsible for exporting the freshly generated Rust code.
 
 {% code title="src/pb/mod.rs" %}
+
 ```rust
 #[path = "eth.erc721.v1.rs"]
 #[allow(dead_code)]
 pub mod erc721;
 ```
+
 {% endcode %}
 
 View this file in the repo by visiting the following link.
 
 [https://github.com/streamingfast/substreams-template/blob/develop/src/pb/mod.rs](https://github.com/streamingfast/substreams-template/blob/develop/src/pb/mod.rs)
+
+### Protobuf & Rust Optional Fields
+
+Fields in Protocol Buffer messages are optional by default. Each field in a Protocol Buffer message has a default value of `None`, indicating that the field has not been populated with any data.
+
+Prost generates Rust code that uses the `Option` type for each field in the Protocol Buffer message. The `Option` type is used to represent the absence of a value in Rust. It allows developers to distinguish between a field that has a value and a field that has not been set. The standard approach to represent nullable data when using Rust is through wrapping optional values in `Option<T>`.
+
```

Comment:
I would quick info here about usual way to consume the `Option`. I would see a quick description for when you are sure it will never be `None` (in this case use `unwrap()`) and otherwise, use a match statement:

```
match person.Location {
  Some(location) => { // Present, do something }
  None => { // Not present, do something }
}
```

Reply:
Here is my proposed update. I'll add it to the file if it sounds good to you.
------------------

Protobufs that are messages define fields within them that can be scalar, or simple, values such as a string or integer. Messages can also contain other messages within them. All messages are processed into Option enums by prost when using Rust. Any scalar values in a message will return the value assigned or an empty string or zero, 0, for number types. Messages nested within messages will be processed into Options. For message types that haven't been assigned a value, the default None for Options that are empty will be used.

Reply:
My comment was about adding a paragraph that explains quickly how to work with `Option` in Rust code. Use `.unwrap` if the dev knows the field is always present, otherwise use:

```
match person.Location { 
    Some(location) => { // Present, do something },
    None => { // Not present, do something }
}
```

Or

```
if let Some(location) = person.location {

}
```

I feel your paragraph is to re-write:

```
For each field that are reference to other Protocol Buffer message types, Prost generates Rust code that uses the `Option` enum. The `Option` enum is used to represent the absence of a value in Rust. It allows developers to distinguish between a field that has a value and a field that has not been set. The standard approach to represent nullable data when using Rust is through wrapping optional values in `Option<T>`
```
?

Reply:
For each field that references other Protocol Buffer message types, `prost` generates Rust code that uses the `Option` enum. The `Option` enum is used to represent the absence of a value in Rust. It allows developers to distinguish between a field that has a value and a field that has not been set. The standard approach to represent nullable data when using Rust is through wrapping optional values in `Option<T>`

Reply:
Please let me know on the slightly modified paragraph above.

Reply:
So the new content is good.

I still want to see a paragraph that gives **quick** instructions on how to deal with `Option` in Rust code, see my previous messages.

Reply:
Ok. Sounds good.

Reply:
Here is the verbiage with code snippets inline. If this is acceptable I'll port the content into the MD file, mark it up, commit the file and close the PR. Please let me know when you have a moment. Thank you.
-------------------------------
The Rust match keyword is used to compare the value of an Option with a Some or None variant. The Rust Option enum is used as the return type for functions, or in this case protobuf-based data fields, that may not always return a value. Protobuf fields that are another nested Message type will be returned as the Option enum type. The following code can be used to check for the existence of a value with match, Option, Some, and None.

match person.Location { 
    Some(location) => { // Present, do something },
    None => { // Not present, do something }
}

Scenarios that are known to always contain a value can use the if let statement and bypass the None arm of the match code shown above. The following code example demonstrates using if let for fields that are known to always contain a value.

if let Some(location) = person.location {
   // A value is present. The processing logic would be written here.
}